### ISSUE TITLE
Translation hackathon

### ISSUE THREAD

**Issue by hturner** [r-devel/rcontribution#36]
This idea was mentioned by @MichaelChirico in the last RCWG meeting. Some possibilities:

1. Run an open event for any language - may need a couple of sessions to cover different time zones.
2. Run a focused event for specific languages. Now that @clente has started working on pt-BR translations, it could be a good time to run an event as a satellite to LatinR. It's a bit sooner than I thought (October 10-14), but maybe it does not take too much time to organize now we have the weblate server for people to use. I know that the maintainer for the Spanish translations would like to pass on this role and would like other people to contribute. Meanwhile there have been various people that have shown interest in Portuguese/Spanish translation in the Collaboration Campfires/on Slack. So it could be a good place to start.

Tagging LatinR folk: @yabellini @rivaquiroga, @natydasilva.
A few others interested in Portuguese/Spanish translation: @beatrizmilz, @rcastelo, @llrs, @eliocamp - not exhaustive!

**Comment by hturner** [comment]
Should also tag @daroczig as the weblate server maintainer and @lawremi as R Core member responsible for translations.

**Comment by eliocamp** [comment]
I think this is a good idea. I participated @MichaelChirico 's translation workshop at useR! (IIRC) and it was super helpful.

**Comment by beatrizmilz** [comment]
@clente

**Comment by hturner** [comment]
At the R Contribution Working we decided to focus on running a satellite event with LatinR first of all (https://latin-r.com/). People interested to get involved in organizing this should join the LatinR Slack.

This might be followed by a joint event with BiocAsia (https://biocasia2022.bioconductor.org/, Dec 1-2, 2022). @kozo2 are you involved with this event already?

**Comment by kozo2** [comment]
@hturner 
Sorry for my late reply.

> 2 hours on a Saturday might be doable for a fair number of people. Do you have a tentative date in mind for the next one?

No, I haven't decided on the date yet.
How about you? (Do you have a tentative date for the next **R Contribution** event?)
If we can have a joint event, I would like to leave it up to you.

> At the R Contribution Working we decided to focus on running a satellite event with LatinR first of all (https://latin-r.com/). People interested to get involved in organizing this should join the LatinR Slack.

I am also a member of [AsiaR](https://github.com/AsiaR-community).
Would it be possible for AsiaR to go with the flow? (cc @jananiravi @adithirgis)
Unlike LatinR, AsiaR has no plans to hold a conference, but does hold meetups.

> This might be followed by a joint event with BiocAsia (https://biocasia2022.bioconductor.org/, Dec 1-2, 2022). @kozo2 are you involved with this event already?

I have heard that BiocAsia will not be holding a workshop style session.
Therefore, it may be difficult to hold a Translation hackathon at BiocAsia. (@PeteHaitch @DarioS @MahShaaban What do you think about this?)
I am involved with BiocAsia, but I will not go to the site (Melbourne) and support it only by updating the website.

**Comment by PeteHaitch** [comment]
@kozo2 @hturner  If there was sufficient interest we might be able to run a translation hackathon in the morning(s) during BioC Asia (the conference is running for 2 afternoons).
Are these largely-online events or do they benefit from having people physically in the same room?
I'm not sure how many multi-lingual people there will be in-person in Melbourne - there will certainly be Chinese-English speakers but I would guess that the number of people who speak a language other than English will be low because many Australians do not have a second language.

**Comment by jananiravi** [comment]
Yes, we can advertise this event on AsiaR once you have tentative dates & languages picked (or even before that to gauge interest). @kozo2 @hturner let us know how you'd like to proceed.

**Comment by MichaelChirico** [comment]
Great news! Happy to support however I can. It would be really great if we can get some quorum to do translations in Arabic and/or Hindi (or any other South Asian language :smiley:)!

re:Weblate -- @daroczig, do you anticipate any problems from having an event with, say 10+ people editing simultaneously? I seem to recall this came up at the RCWG meeting but I'm forgetting the details.

**Comment by kozo2** [comment]
@PeteHaitch 
> Are the hackathons largely-online events or do they benefit from having people physically in the same room?

Yes, I think the hackathons will be largely-online events.
No, they don't. As you mentioned later, I don't think local participants are very interested in multilingualization.

**Comment by daroczig** [comment]
@MichaelChirico the weblate server should be able to handle that amount of traffic without any problem, but just in case, I'm happy to bump the instance size for the event to make sure to have some extra resources

**Comment by kozo2** [comment]
@daroczig @MichaelChirico 
Can I see how the Weblate server is used in your project?
I'm not using Weblate, I'm using Crowdin. (https://bioconductor.crowdin.com/)

**Comment by daroczig** [comment]
@kozo2 sure, you can find the weblate server at https://translate.rx.studio

I picked weblate as it is self-hosted, and I recall some discussions on R-devel about preferring self-hosted solutions instead of cloud services, see e.g. GitHub VS Subversion. This way we have full control over the translations and workflows, although at the expense of maintaining it.

On the other hand, I'm totally open to other solutions as well - depending on what the community's and R Core's needs are.

**Comment by kozo2** [comment]
@daroczig Thank you for information.
I understand what you mean and what features Weblate has.
For our purposes, it seems appropriate to use Crowdin, so I will continue to use it.

**Comment by hturner** [comment]
@PeteHaitch @kozo2 thanks for your comments. It sounds like there isn't much benefit tying in with BiocAsia and it may be better to partner with AsiaR for broader reach. Since we are focusing on LatinR first in October, maybe aim for mid/late November? We could transfer discussion/organization to the AsiaR Slack if everyone interested is a member there (or is willing to join: https://github.com/asiar-community).

**Comment by hturner** [comment]
@Athene-ai it looks like your languages may be English and Italian? If that's right, Italian is quite well-maintained over by Daniele Medri over here: https://github.com/dmedri/R-italian-lang. It looks like there is no work to do there, but you might contact him to see if there is a way you can help (see https://developer.r-project.org/TranslationTeams.html).

**Comment by Athene-ai** [comment]
@hturner I have just contacted Daniele Medri ;-)

**Comment by dmedri** [comment]
@Athene-ai you're welcome. I just sent a reply to your email...

**Comment by Athene-ai** [comment]
@dmedri  perfect ;-)

**Comment by mmaechler** [comment]
> .. you can find the weblate server at https://translate.rx.studio

well, how on earth should R Foudnation / R project parts be part of "rstudio" ???
Yes, I agree  @daroczig  that self hosting is a good idea, and in particular  for our translation projects be hosted on weblate.

But honestly,  the URL containing  'rstudio' is an absolute no go!

**Comment by daroczig** [comment]
Sorry for the confusion, @mmaechler -- I think you were not there when we discussed that I am happy to migrate the Weblate server under the r-project.org domain and/or host on R Project servers. I decided to put it under the rx.studio (please note it's not RStudio) domain and AWS account, as I had resources there and did not want to wait for R Core/Foundation to dedicate resources to an experimental project.

So to keep it short, if someone can help me with the below:

1. Get a subdomain under r-project, e.g. translate.r-project.org or i18n.r-project.org or similar
2. Provide a server for running Weblate (currently it runs on a virtual machine with 8 gigs of RAM using Docker)

Then I am happy to migrate the service, and I can keep maintaining or help others do it if that's preferred.
